src/lib.rs

```diff
@@ -156,6 +156,51 @@ impl Vob<usize> {
         v.mask_last_block();
         v
     }
+
+    /// Create a Vob from a `u8` slice. The most significant bit of each byte comes first in the
+    /// resulting Vob.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// use vob::Vob;
+    /// fn main() {
+    ///     let v = Vob::from_bytes(&[0b10100000, 0b00010010]);
+    ///     assert_eq!(v, vob![true, false, true, false, false, false, false, false,
+    ///                        false, false, false, true, false, false, true, false]);
+    /// }
+    /// ```
+    pub fn from_bytes(slice: &[u8]) -> Vob<usize> {
+        let mut v = Vob::with_capacity(slice.len()
+                                            .checked_mul(8)
+                                            .expect("Overflow detected"));
+        for i in 0..blocks_required::<usize>(slice.len() * 8) {
+            let mut w = usize::zero(); // word
+            for j in 0..bytes_per_block::<usize>() {
+                let off = i * bytes_per_block::<usize>() + j;
+                if off >= slice.len() {
+                    // If slice doesn't contain a whole number of words, we'll end up with one
+                    // block at the end with "missing" bytes; since these are equivalent to 0, we
+                    // can simply ignore the fact they're missing.
+                    continue;
+                }
+                let b = slice[off];
+                if b != 0 {
+                    let mut rb: u8 = 0; // the byte b with its bits in reverse order
+                    for k in 0..8 {
+                        if b & (1 << k) != 0 {
```

Comment:
This conditional is designed to avoid arithmetic and shifts if there is no need to flip a bit on?

Because otherwise, I think it's equivalent to:
```
for k in 0..8 {
    rb |= b & (1 << k) << (8 - k - 1);
}
```

Reply:
Entirely up to you. Maybe benchmark to see if one is faster?

Reply:
It should be `rb |= ((b >> k) & 1) << (8 - k - 1);`. It's still going to be pretty fast, and it's shorter, so I like it.